### PR TITLE
Improve POST response

### PR DESCRIPTION
**Pull request by jiah29**
#19 

## Context

The previous POST response body is ugly and hard to parse. I have modified the ServerMainEndpointHandler class so that it returns a nicer response body.

## Your Changes

Modify and fully document ServerMainEndpointHandler, RootEndpointHandler and RestServiceCorsApplications. Note that the ServerMainEndpointHandler does not use the HttpResponseMain class because it is actually the class that made the return format really tricky to parse. (We might need to consider not using the HttpResponseMain class for the database endpoint as well).

**Type of change** (select all that apply):

- [x] New feature (additional functionality separate from existing code)
- [x] Refactoring (changing code structure only)
- [x] Documentation update (adding more comments)
- [x] Bug fix (non-breaking change that solves an error/issue)
- [x] User Interface change (design changes)
- [x] Test update (adding test code)

## Testing

Completed a full test suite for ServerMainEndpointHandler.java, and manually tested the post response formatting on postman.

## Checklist

- [x] Code is up to date with main branch (no merge conflicts)
- [x] Code has been properly tested
- [x] Code has been properly documented
- [x] One or two people have reviewed this pull request

### CODE REVIEW

**Comment by jiah29** [comment]
Approve and merge https://github.com/autotrader-plus/autotrader-plus/pull/17 first! Very Important!

**Review by parthaba — APPROVED** [review]
Changes look good and tests succeed

**Review by Danx1129 — APPROVED** [review]
Looks gooder
